### ISSUE TITLE
Chat Processor moved to disabled plugins

### ISSUE THREAD

**Issue by zo6zo6** [Kxnrl/Store#109]
by installing the store on my live server, upon first startup chat processor will be moved to disabled. why is this happening and how can it be fixed. i have other plugins on the server that depends on chat processor to work

**Comment by Kxnrl** [comment]
store has a built-in chat-processor, provide chat-processor api.

**Comment by zo6zo6** [comment]
what can i do to fix this

**Comment by Kxnrl** [comment]
can u share the gloves plugin github page here?

**Comment by zo6zo6** [comment]
yeah sure. its this one https://github.com/Franc1sco/Franug-CSGO-Gloves
OR this one https://github.com/kgns/gloves 
i can use any on these two

**Comment by zo6zo6** [comment]
in the alliedmodders thread of the gloves plugin the author said this 
"If you are using Custom Player OR Arms Models on your server, you have to modify the plugin that does the model changing, and make it use Gloves_SetArmsModel native function instead. Or your players arms will be invisible if they do not use any gloves."

**Comment by Kxnrl** [comment]
emmm, did u mean ArmsFix?

**Comment by zo6zo6** [comment]
i will upload a video tomorrow to help you understand better

**Comment by Kxnrl** [comment]
ok

**Comment by zo6zo6** [comment]
here is the video.
i explain in the ingame chat the problem and show it on the video

https://youtu.be/jOi3k0S4K8Q

**Comment by Kxnrl** [comment]
```sourcepawn
Store_RemoveClientGloves
```
Just comment this line in store/module/skin.sp

**Comment by zo6zo6** [comment]
Store_RemoveClientGloves appears 5 times in store/module/skin.sp in the following lines
Line 240:  Store_RemoveClientGloves(client, 0);
Line 265:  if(strlen(arms) > 3) Store_RemoveClientGloves(client, 0);
Line 337:   Store_RemoveClientGloves(client, 0);
Line 667: void Store_RemoveClientGloves(int client, int m_iData = -1)
Line 734: Store_RemoveClientGloves(client, 0);

i commended lines 240, 337 and 734 but it did not work. 
Which one should i comment to make it work?

**Comment by Kxnrl** [comment]
still using default arms?

**Comment by zo6zo6** [comment]
> still using default arms?

yes. same behavior, as shown in the video

**Comment by Kxnrl** [comment]
I will add 1 forwards , 
```sourcepawn
Action Store_OnSetPlayerSkinArms(int client, const char[] arms)

**Comment by zo6zo6** [comment]
> I will add 1 forwards ,
> 
> ```sourcepawn
> Action Store_OnSetPlayerSkinArms(int client, const char[] arms)
> ```

ok, will it be working is the next version of the store?

**Comment by zo6zo6** [comment]
i installed the new version and it did not fix the problem. what can i do now?

**Comment by Kxnrl** [comment]
add this code into your plugin
```sourcepawn
public Action Store_OnSetPlayerSkinArms(int client, const char[] arms)
{
    if (CheckPlayerHasGloves(client)) // this check define in your glove plugin
    {
        return Plugin_Handled;
    }
}

**Comment by zo6zo6** [comment]
> add this code into your plugin
> 
> ```sourcepawn
> public Action Store_OnSetPlayerSkinArms(int client, const char[] arms)
> {
>     if (CheckPlayerHasGloves(client)) // this check define in your glove plugin
>     {
>         return Plugin_Handled;
>     }
>     return Plugin_Continue;
> }
> ```

tried to add it on gloves plugin and it gave me the following error

> gloves.sp(184) : error 017: undefined symbol "CheckPlayerHasGloves"

**Comment by Kxnrl** [comment]
> ```sourcepawn
> // this check define in your glove plugin
> ```

// this check define in your glove plugin

**Comment by zo6zo6** [comment]
> > ```sourcepawn
> > // this check define in your glove plugin
> > ```
> 
> // this check define in your glove plugin

i dont know what am i supposed to do with that. all compile attempts fail

**Comment by zo6zo6** [comment]
i contacted the author of the gloves ugin and he said:
"If you are using Custom Player OR Arms Models on your server, you have to modify the plugin that does the model changing, and make it use Gloves_SetArmsModel native function instead. Or your players arms will be invisible if they do not use any gloves."

he also said

"tell the author of your store plugin to check this plugins natives and add support for this gloves plugin if he wants gloves to work. there are natives released just for this problem"

the gloves plugin i am using is the following

https://github.com/kgns/gloves

do you think you can fix this problem?

**Comment by mgeerse** [comment]
I also need chat processor to **work** with other plugins, you mention an api @Kxnrl but it's nowhere to be found. Plugins break because of this 'feature'.
